### ISSUE TITLE
FSE: Pages with autosaves cannot be previewed [2]

### ISSUE THREAD

**Issue by apeatling** [Automattic/wp-calypso#35484]
The easiest way to reproduce this is to preview any page one time, since this will trigger an autosave (but I don’t understand why, there are no changes in the content. This also makes the issue with the revisions above more noticeable). 

The second time you try to preview a page, the modal will show a gray background.

**Comment by glendaviesnz** [comment]
Initial investigation of this shows that it is not FSE site specific. The same error seems to occur on any Simple or Atomic site when following the above steps.

For some reason the /wp-calypso/apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js is sending a null preview url if a page has an autosave. Time to work out how to debug the iframe-bridge-server so I can work out why this is.

**Comment by glendaviesnz** [comment]
I suspect it it something to do with the following in /wp-content/plugins/gutenberg/packages/editor/src/store/reducer.js

```
case 'REQUEST_POST_UPDATE_START':
	// Invalidate known preview link when autosave starts.
	if ( state && action.options.isPreview ) {
		return null;
	}
```

as in some instances (but not all it seems) related to autosaves the getEditedPostPreviewLink selector returns null, because the editorState.previewLink is null. Still trying to work out why that is, but if that situation rings any bells with you let me know.

**Comment by glendaviesnz** [comment]
some potentially useful information here which I haven't had chance to read through yet https://github.com/WordPress/gutenberg/issues/7179

**Comment by glendaviesnz** [comment]
Calypsoify is definitely still displaying the behaviour noted here - https://github.com/WordPress/gutenberg/issues/7852#issuecomment-413049850 which is supposed to have been fixed in https://github.com/WordPress/gutenberg/pull/7945. I can't replicate the issue in block editor in wp-admin, only in calypsoify - which is running a version of the editor that should include the above fix. Will do some more digging tomorrow.

**Comment by glendaviesnz** [comment]
Duplicate issue - https://github.com/Automattic/wp-calypso/issues/33799

**Comment by glendaviesnz** [comment]
fixed by #35824
